tests/Core/Generators/GeneratorTest.php

```diff
@@ -0,0 +1,240 @@
+        $multidocExpected .= "Two Standard Blocks, No Code\n";
+        $multidocExpected .= "Two Standard Blocks, One Code Comparison\n";
+        $multidocExpected .= "Two Standard Blocks, Three Code Comparisons\n";
+
+        return [
+            'Standard without docs'            => [
+                'standard' => __DIR__.'/NoDocsTest.xml',
+                'expected' => '',
+            ],
+            'Standard with one doc file'       => [
+                'standard' => __DIR__.'/OneDocTest.xml',
+                'expected' => "One Standard Block, No Code\n",
+            ],
+            'Standard with multiple doc files' => [
+                'standard' => __DIR__.'/StructureDocsTest.xml',
+                'expected' => $multidocExpected,
+            ],
+        ];
+
+    }//end dataGeneratingDocs()
+
+
+    /**
+     * Test that the documentation for each standard passed on the command-line is shown separately.
+     *
+     * @covers \PHP_CodeSniffer\Runner::runPHPCS
+     *
+     * @return void
+     */
+    public function testGeneratorWillShowEachStandardSeparately()
+    {
+        $standard        = __DIR__.'/OneDocTest.xml';
+        $_SERVER['argv'] = [
+            'phpcs',
+            '--generator=Text',
+            "--standard=$standard,PSR1",
+            '--report-width=80',
+        ];
+
+        $regex = '`^
+            \R*                                                      # Optional blank line at the start.
+            (?:
+                (?P<delimiter>-+\R)                                  # Line with dashes.
+                \|[ ]GENERATORTEST[ ]CODING[ ]STANDARD:[ ][^\|]+\|\R # Doc title line with prefix expected for first standard.
+                (?P>delimiter)                                       # Line with dashes.
+                .+?\R{2}                                             # Standard description.
+            )                                                        # Only expect this group once.
+            (?:
+                (?P>delimiter)                                       # Line with dashes.
+                \|[ ]PSR1[ ]CODING[ ]STANDARD:[ ][^\|]+\|\R          # Doc title line with prefix expected for second standard.
+                (?P>delimiter)                                       # Line with dashes.
+                .+?\R+                                               # Standard description.
+                (?:
+                    -+[ ]CODE[ ]COMPARISON[ ]-+\R                    # Code Comparison starter line with dashes.
+                    (?:.+?(?P>delimiter)\R){2}                       # Arbitrary text followed by a delimiter line.
+                )*                                                   # Code comparison is optional and can exist multiple times.
+                \R+
+            ){3,}                                                    # This complete group should occur at least six times.
```

Comment:
If I'm reading this correctly, the regex expects the group to match at least three times, and the comment mentions it expects at least six times. On a quick look, I'm seeing seven doc blocks for PSR1 standards when running `bin/phpcs --standard=tests/Core/Generators/OneDocTest.xml,PSR1 --generator=Text`:

<details>
  <summary>Command output</summary>

  ```
$ bin/phpcs --standard=tests/Core/Generators/OneDocTest.xml,PSR1 --generator=Text         

--------------------------------------------------------------
| GENERATORTEST CODING STANDARD: ONE STANDARD BLOCK, NO CODE |
--------------------------------------------------------------

Documentation contains one standard block and no code comparison.


------------------------------------------
| PSR1 CODING STANDARD: BYTE ORDER MARKS |
------------------------------------------

Byte Order Marks that may corrupt your application should not be used.  These include 0xefbbbf
(UTF-8), 0xfeff (UTF-16 BE) and 0xfffe (UTF-16 LE).


----------------------------------------
| PSR1 CODING STANDARD: CONSTANT NAMES |
----------------------------------------

Constants should always be all-uppercase, with underscores to separate words.

----------------------------------------- CODE COMPARISON ------------------------------------------
| Valid: All uppercase constant name.            | Invalid: Mixed case or lowercase constant       |
|                                                | name.                                           |
----------------------------------------------------------------------------------------------------
| define('FOO_CONSTANT', 'foo');                 | define('Foo_Constant', 'foo');                  |
|                                                |                                                 |
| class FooClass                                 | class FooClass                                  |
| {                                              | {                                               |
|     const FOO_CONSTANT = 'foo';                |     const foo_constant = 'foo';                 |
| }                                              | }                                               |
----------------------------------------------------------------------------------------------------


---------------------------------------------------
| PSR1 CODING STANDARD: ALTERNATIVE PHP CODE TAGS |
---------------------------------------------------

Always use <?php ?> to delimit PHP code, do not use the ASP <% %> style tags nor the <script
language="php"></script> tags. This is the most portable way to include PHP code on differing
operating systems and setups.


---------------------------------------
| PSR1 CODING STANDARD: PHP CODE TAGS |
---------------------------------------

Always use <?php ?> to delimit PHP code, not the <? ?> shorthand. This is the most portable way to
include PHP code on differing operating systems and setups.


-------------------------------------------
| PSR1 CODING STANDARD: CLASS DECLARATION |
-------------------------------------------

Each class must be in a file by itself and must be under a namespace (a top-level vendor name).

----------------------------------------- CODE COMPARISON ------------------------------------------
| Valid: One class in a file.                    | Invalid: Multiple classes in a single file.     |
----------------------------------------------------------------------------------------------------
| <?php                                          | <?php                                           |
| namespace Foo;                                 | namespace Foo;                                  |
|                                                |                                                 |
| class Bar {                                    | class Bar {                                     |
| }                                              | }                                               |
|                                                |                                                 |
|                                                | class Baz {                                     |
|                                                | }                                               |
----------------------------------------------------------------------------------------------------

----------------------------------------- CODE COMPARISON ------------------------------------------
| Valid: A vendor-level namespace is used.       | Invalid: No namespace used in file.             |
----------------------------------------------------------------------------------------------------
| <?php                                          | <?php                                           |
| namespace Foo;                                 | class Bar {                                     |
|                                                | }                                               |
| class Bar {                                    |                                                 |
| }                                              |                                                 |
----------------------------------------------------------------------------------------------------


--------------------------------------
| PSR1 CODING STANDARD: SIDE EFFECTS |
--------------------------------------

A PHP file should either contain declarations with no side effects, or should just have logic
(including side effects) with no declarations.

----------------------------------------- CODE COMPARISON ------------------------------------------
| Valid: A class defined in a file by itself.    | Invalid: A class defined in a file with other   |
|                                                | code.                                           |
----------------------------------------------------------------------------------------------------
| <?php                                          | <?php                                           |
| class Foo                                      | class Foo                                       |
| {                                              | {                                               |
| }                                              | }                                               |
|                                                |                                                 |
|                                                | echo "Class Foo loaded."                        |
----------------------------------------------------------------------------------------------------


-------------------------------------
| PSR1 CODING STANDARD: METHOD NAME |
-------------------------------------

Method names MUST be declared in camelCase.

----------------------------------------- CODE COMPARISON ------------------------------------------
| Valid: Method name in camelCase.               | Invalid: Method name not in camelCase.          |
----------------------------------------------------------------------------------------------------
| class Foo                                      | class Foo                                       |
| {                                              | {                                               |
|     private function doBar()                   |     private function do_bar()                   |
|     {                                          |     {                                           |
|     }                                          |     }                                           |
| }                                              | }                                               |
----------------------------------------------------------------------------------------------------
```
</details>

Updating the number of minimum expected matches in the regex to six or seven causes the test to fail.

Reply:
> If I'm reading this correctly, the regex expects the group to match at least three times, and the comment mentions it expects at least six times. On a quick look, I'm seeing seven doc blocks for PSR1 standards when running `bin/phpcs --standard=tests/Core/Generators/OneDocTest.xml,PSR1 --generator=Text`:
> 
> Updating the number of minimum expected matches in the regex to six or seven causes the test to fail.

... which is why it was set to `{3,}`

The regex part for matching the code comparison is a bit hefty what with the use of `.`, so if set to `{6,}`, the regex easily hits a catastrophic backtrace error (in part depends on the PHP version).

I'll update the comment though ;-)

Reply:
Thanks for the additional details.